### PR TITLE
[mlir][vector] Update document for `vector.splat`(NFC)

### PR DESCRIPTION
**Pull request by CoTinker**
This PR updates the document for `vector.splat`, specifying that the operand type must match the element type of the result.

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
Thanks, good catch!

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td [hunk 1]
```diff
@@ -2807,8 +2807,8 @@ def Vector_SplatOp : Vector_Op<"splat", [
     Example:
 
     ```mlir
-    %s = arith.constant 10.1 : f32
-    %t = vector.splat %s : vector<8x16xi32>
+    %s = arith.constant 1.0 : f32
```
Comment (banach-space) [inline mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:2810]: [nit] This change is unrelated

**Review by banach-space — APPROVED** [review]
LGTM % one nit

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td (hunk 1 above)
Comment (CoTinker) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td]: Done

**Comment by CoTinker** [comment]
Thanks for your review.

**Review by dcaballe — APPROVED** [review]
